### ISSUE TITLE
Garage door controller not updating status correctly

### ISSUE THREAD

**Issue by yhaugom** [athombv/com.aeotec#172]
My garage door controller is not updating it's status correctly. 
When closing it "manually" (by remote), Homey doesn't get updated. 
I have a flow triggering a countdown to close the door automatically after a few minutes. 
This flow thinks the door is open when it is not. But when it "fires", nothing happens to the door. Other than the status updating in the app.
When trying to open/close the door in the app, I (almost) always have to push twice to make it do what I intended.
I have the flip open/close option toggled in app settings.
I can get you z-wave packet data between it and Homey.

**Comment by yhaugom** [comment]
I will send you a support ticket containing the actual communication between Homey and the garage door controller, and explanation of what I am doing prior to each step of the communication. I will reference this issue in the ticket.

**Comment by jpmbosco** [comment]
I have exactly the same issue since upgrade to Homey v2. I added 2 flows, 1 to announce when the door closed and 1 to do the same when open. The flow flashes the ring on Homey and says door open or door closed. Homey always announces correctly when I use my garage fob and as the previous comments, does not close or open correctly using either app or flows. This worked prior to v2 upgrade.

**Comment by jpmbosco** [comment]
Hi yhaugom, did you get your issue resolved?. I have been able to get round the issue as you commented that you have to push twice to get the door to ope/close, I edited my flow to include 2 door closes to get the door to close after a period of time an this appears to work.

**Comment by yhaugom** [comment]
@jpmbosco No, I didn't get it resolved. Noticed they changed some small things in the driver in the alpha branch. Dont want to fix this by the "hack" you mentioned, but it could be a temporary solution. Wouldn't keep auto update on then, cause then, if it is suddenly fixed, you will have a problem again.
I can handle my situation for a short period of time by checking trough my surveilance camera if the door is actually closed or not... But really want this fixed obviously...
Thanks for the suggestion.

**Comment by osochor** [comment]
Here the same!!!
Homey 2.0.2
Aeotec App 2.0.18

**Comment by jpmbosco** [comment]
Hi yhaugom, I have emailed Athom again today to ask what is being done to fix this defect as I have heard nothing back from either Athom or Aeotec.

**Comment by yhaugom** [comment]
Aeotec has nothing to do with this I think. They made the product for sure, but it is up to Athom to communicate with the device correctly. I can see in the logs that the controller is sending the correct information, but the status is not updated in the Homey app.
First when I hit open, the garage door controller sends "State": "Opening", then when it is open it sends "State": "Open", but the app then says closed. And so on...

**Comment by macca73** [comment]
Hi, I have the same issue as well.  When i want to open my garage door via Homekit, I say “open garage door”. When I want to close it I have to say “open garage door”.....haha.  My other half thinks I’m crazy!  Hope it gets resolved soon.

**Comment by yhaugom** [comment]
I tested [this commit](https://github.com/athombv/com.aeotec/commit/054126e1ab13dce39c21fff373d034676d0a5365) but it doesnt solve anything. I think that flipping the set value is causing problems. But if I understand this correctly the flip setting doesn't make any sense.
In my opinion this hack should be removed at once. The app is all ready in a state that can't be trusted (regarding the garage door opener), so it would be much better to make it right and just get rid of the flip setting.
I just testet the following code, and it seems to work much better. Please consider this, cause this code/behaviour is really confusing. It make little sense to people to go to settings and flip the logic to make their garage door behave correctly. And as it is now, it doesn't behave as expected either way.

`onMeshInit() { this.registerCapability('locked', 'BARRIER_OPERATOR', { get: 'BARRIER_OPERATOR_GET', report: 'BARRIER_OPERATOR_REPORT', reportParser: report => { return report.State === 'Closed'; }, set: 'BARRIER_OPERATOR_SET', setParser: (input) => { return {'Target Value': (input) ? 'CLOSE' : 'OPEN'}; }, }); }`
